query/src/main/scala/filodb/query/exec/rangefn/RangeFunction.scala

```diff
@@ -286,17 +287,18 @@ object RangeFunction {
   /**
    * Returns a function to generate a ChunkedRangeFunction for Long columns
    */
-  def longChunkedFunction(dataset: Dataset,
+  def longChunkedFunction(schema: ResultSchema,
                           func: Option[RangeFunctionId],
                           funcParams: Seq[Any] = Nil): RangeFunctionGenerator = {
+    val isDownsampleGauge = (schema.columns == Schemas.dsGauge.dataInfos)
```

Comment:
As mentioned earlier, the resultSchema would just be the columns necessary for executing the downsample query. This would not necessarily be the same as full dsGuage schema.